plugin/json-message-codec.go

```diff
@@ -0,0 +1,20 @@
+package plugin
+
+import (
+	"encoding/json"
+)
+
+// JSONMessageCodec implements a MessageCodec using JSON for message encoding.
+type JSONMessageCodec struct{}
```

Comment:
I deliberately left out the JSONMessageCodec because in Go its better to decode directly to known structs. Perhaps the encoding part is useful,  simplifies returning values, but the decode part could simply return the bytes and leave actual decoding up to the method implementation.